firrtl/src/main/scala/firrtl/ir/Serializer.scala

```diff
@@ -259,14 +271,14 @@ object Serializer {
           readwriters,
           readUnderWrite
         ) =>
-      b ++= "mem "; b ++= name; b ++= " :"; s(info); newLineAndIndent(1)
+      b ++= "mem "; b ++= legalize(name); b ++= " :"; s(info); newLineAndIndent(1)
       b ++= "data-type => "; s(dataType); newLineAndIndent(1)
       b ++= "depth => "; b ++= depth.toString(); newLineAndIndent(1)
       b ++= "read-latency => "; b ++= readLatency.toString; newLineAndIndent(1)
       b ++= "write-latency => "; b ++= writeLatency.toString; newLineAndIndent(1)
-      readers.foreach { r => b ++= "reader => "; b ++= r; newLineAndIndent(1) }
-      writers.foreach { w => b ++= "writer => "; b ++= w; newLineAndIndent(1) }
-      readwriters.foreach { r => b ++= "readwriter => "; b ++= r; newLineAndIndent(1) }
+      readers.foreach { r => b ++= "reader => "; b ++= legalize(r); newLineAndIndent(1) }
+      writers.foreach { w => b ++= "writer => "; b ++= legalize(w); newLineAndIndent(1) }
+      readwriters.foreach { r => b ++= "readwriter => "; b ++= legalize(r); newLineAndIndent(1) }
       b ++= "read-under-write => "; b ++= readUnderWrite.toString
     case Attach(info, exprs) =>
       // exprs should never be empty since the attach statement takes *at least* two signals according to the spec
```

Comment:
What about verification Statements? They have labels which are names in the sense that they can be annotated and they should correspond to labels in the Verilog.

Reply:
Good catch. I had considered this and thought it wasn't necessary. However, it is. I added a commit that makes these literal identifiers and adds checks that it works here: https://github.com/chipsalliance/chisel/pull/3374/commits/9587c6c0925851220d1b4a2f9254fd6849331a4e